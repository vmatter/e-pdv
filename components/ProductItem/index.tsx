import { useShoppingCart, formatCurrencyString } from 'use-shopping-cart';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import { Product } from '../Products';

type Props = {
  product: Product;
};

const ProductItem = ({ product }: Props) => {
  const { addItem, removeItem } = useShoppingCart();
  return (
    <Card>
      <CardActionArea>
        <CardMedia
          component="img"
          alt={product.name}
          image={product.image}
          height="140"
          width="140"
        />
        <CardContent>
          <Typography gutterBottom variant="h5" component="h2">
            {product.name}
          </Typography>
          <Typography component="p">
            {formatCurrencyString({
              value: product.price,
              currency: product.currency,
            })}
          </Typography>
        </CardContent>
      </CardActionArea>
      <CardActions>
        <Button
          size="small"
          color="primary"
          onClick={() => removeItem(product.sku)}
        >
          Remove
        </Button>
        <Button size="small" color="primary" onClick={() => addItem(product)}>
          Add Item
        </Button>
      </CardActions>
    </Card>
  );
};

// <Product key={product.sku}>
//   <ProductImg src={product.image} alt={product.name} />
//   <h2>{product.name}</h2>
//   <p className="price">
//     {formatCurrencyString({
//       value: product.price,
//       currency: product.currency,
//     })}
//   </p>
//   <button
//     className="cart-style-background"
//     onClick={() => addItem(product)}
//   >
//     Add to cart
//   </button>
//   <button
//     className="cart-style-background"
//     onClick={() => removeItem(product.sku)}
//   >
//     Remove
//   </button>
// </Product>
export default ProductItem;
