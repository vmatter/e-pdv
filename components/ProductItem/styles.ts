import styled from '@emotion/styled';
import Typography from '@material-ui/core/Typography';

export const ProductName = styled(Typography)`
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
`;
