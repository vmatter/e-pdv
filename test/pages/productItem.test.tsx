import React from 'react';
import { render } from '../testUtils';
import '@testing-library/jest-dom';
import ProductItem from '../../components/ProductItem';

describe('Index Page', () => {
  const expectedProps = {
    name: 'Notebook Dell XPS',
    description: 'Notebook Dell XPS',
    sku: 'sku_notebook',
    price: 540000,
    image:
      'https://images.unsplash.com/photo-1593642532744-d377ab507dc8?ixid=MXwxMjA3fDF8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=750&q=80',
    currency: 'BRL',
  };

  test('should render name, and imagem', async () => {
    const { getByText } = render(<ProductItem product={expectedProps} />);
    const name = getByText(expectedProps.name);

    expect(name).toBeVisible();
  });

  test('matches snapshot', () => {
    const { asFragment } = render(<ProductItem product={expectedProps} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
