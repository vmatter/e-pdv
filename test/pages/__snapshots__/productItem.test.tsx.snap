// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
      tabindex="0"
      type="button"
    >
      <img
        alt="Notebook Dell XPS"
        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
        height="140"
        src="https://images.unsplash.com/photo-1593642532744-d377ab507dc8?ixid=MXwxMjA3fDF8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=750&q=80"
        width="140"
      />
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-17fzm4y-MuiTypography-root"
        >
          Notebook Dell XPS
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1m0ae9b-MuiTypography-root"
        >
          R$5,400.00
        </p>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <div
      class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
    >
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root css-a529o-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
      >
        <span
          class="MuiButton-label css-8xplcm-MuiButton-label"
        >
          Remove
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root css-a529o-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
      >
        <span
          class="MuiButton-label css-8xplcm-MuiButton-label"
        >
          Add Item
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
